### PR TITLE
Update and improve the CONTRIBUTING.md guidelines

### PR DESCRIPTION
**Pull request by simonbasle**

### CODE REVIEW

**Comment by simonbasle** [comment]
@reactor/team I've update the wording, fixes some outdated instructions and added a few elements.

As discussed previously, I've also clarified the use of the Gitter channels (and removed reference to the core one, as we plan on archiving it).

@violetagg I've taken inspiration from reactor-netty's CONTRIBUTING.md regarding the CLA section. I've also added a "Super Short Version" at the beginning for those who want to go fast fast fast 😆

As a result, I think `reactor-netty/CONTRIBUTING.md` can and should be removed so we have a single place for guidelines. wdyt?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -3,15 +3,32 @@
...
+## :runner: Super Short Version
+
+1. Work from a feature branch
+ - Evaluate or ask the team whether the fix should be done in oldest maintenance branch (if there are any active, eg. `3.3.x`)
+2. All contributions go through a [Pull Request](#mortar_board-learning-more-about-the-tools-we-use) and code review process, including our own!
+3. Make sure to sign the [CLA](#clipboard-contributor-license-agreement)
+4. Include tests
+5. Update [license headers](#license-headers) if necessary
+6. Make sure commits are focused on a single issue, ideally within a dedicated PR
+ - Before merging, we'll strive to have single-concern commit(s) with a [well crafted message]((#black_nib-commit-message-convention)
```
Comment (violetagg) [inline CONTRIBUTING.md:21]:
```suggestion
 - Before merging, we'll strive to have single-concern commit(s) with a [well crafted message](#black_nib-commit-message-convention)
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -154,20 +195,37 @@ maintenance branche(s).
...
+will be triaged into `3.3.x Backlog`).
 
 If on the other hand the issue should only be fixed in the next feature release,
-it will be affected to the `Backlog` triage milestone and should be developed
-against `main`.
+assuming `3.4.x` is the current main line it will be affected to the `3.4.x Backlog`
+triage milestone and should be developed against `main`.
+
+### From PR to `main`
+For each PR, there will generally be **a single commit** that goes into `main` (or formerly `master`). We will try to use the `squash-and-merge` strategy on GitHub to that effect, but if you do that yourself once the PR is approved, it's even better :heart:
```
Comment (violetagg) [inline CONTRIBUTING.md:207]:
```suggestion
For each PR, there will generally be **a single commit** that goes into `main` (or formerly `master`). We will try to use the `squash-and-merge` strategy on GitHub to that effect.

Comment (violetagg) [reply CONTRIBUTING.md:207]: We discussed that we want to preserve the history so it is better that the `squash-and-merge` is done only by maintainers via the GitHub UI

**Review by violetagg — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (simonbasle) [reply CONTRIBUTING.md:207]: I'm not sure if you're referring to preserving the history of production branches or to the fact that force-pushes in PRs makes reviewing harder for us? in any case I'm ok with removing that sentence: if we want the contributor to do some polishing of their PR branch before merging, we can always ask them on a case by case basis.

Comment (violetagg) [reply CONTRIBUTING.md]: yeah we wanted to not have force pushes if that's possible

**Review by violetagg — APPROVED** [review]

**Pull request merged**